core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -49,29 +48,28 @@ class Project(
   private val resolver = new SourceResolver(config)
   private val searchService = new SearchService(config, resolver)
 
-  private case object AskReTypeCheck
+  private case class AskReTypeCheck(projects: List[EnsimeProjectId])
 
   // map to get all projects that list the key project in 'depends'
   private val dependentProjects: Map[EnsimeProjectId, List[EnsimeProjectId]] =
     config.projects.map(
       p => p.id -> config.projects.filter(_.depends.contains(p.id)).map(_.id)
     ).toMap
 
-  val restartProjects: mutable.Queue[List[EnsimeProjectId]] = mutable.Queue.empty
   // FileChangeListener for ClassFileWatcher
   private val reTypecheck = new FileChangeListener {
+    private val projectActor = self
     private val askReTypeCheck =
       Debouncer.forActor(
-        self,
-        AskReTypeCheck,
+        projectActor,
+        RestartScalaCompilerReq(None, ReloadStrategy.KeepLoaded),
         delay = (5 * Timing.dilation).seconds,
         maxDelay = (20 * Timing.dilation).seconds
       )
     def fileChanged(f: FileObject): Unit = {
       val projectId = config.findProject(f)
       projectId foreach { projectId =>
-        restartProjects.enqueue(projectId :: dependentProjects.getOrElse(projectId, Nil))
-        askReTypeCheck.call()
+        projectActor ! AskReTypeCheck(projectId :: dependentProjects.getOrElse(projectId, Nil))
```

Comment:
Yeah, I realise that. The roadblock was not being able to send any parameters to call method hence came up with the initial mutable queue.